bin/benchmark_nc.py

```diff
@@ -0,0 +1,601 @@
+        output.write('"%s",%f,%f\n' % (model_data['name'], model_data['mean'], model_data['stddev']))
+    output.flush()
+    output.close()
+    print("Wrote as %s" % output.name)
+
+def ingest_csv(datasets=None, range=None):
+    existing_files = filter(lambda x: os.path.isfile(x[1]), datasets)
+    assert len(datasets) == len(existing_files)
+
+    if range:
+        range = map(lambda x: int(x), range.split(','))
+
+    datas = {}
+    for (dsname, dsfile) in datasets:
+        print('Reading %s from %s' % (dsname, dsfile))
+        with open(dsfile) as f:
+            d = csv.DictReader(f)
+            datas[dsname] = []
+            for e in d:
+                if range:
+                    re       = reduce_filename(e['model'])
+                    in_range = (re >= range[0] and re <= range[1])
+                    if in_range:
+                        datas[dsname].append(e)
+                else:
+                    datas[dsname].append(e)
+
+    return datas
+
+def produce_plot(input=None, output=None):
+    x = range(len(input))
+    xlabels = map(lambda a: a['name'], input)
+    y = map(lambda a: a['mean'], input)
+    yerr = map(lambda a: a['stddev'], input)
+
+    print('y=', y)
+    print('yerr=', yerr)
+    plt.errorbar(x, y, yerr=yerr)
+    plt.show()
+    print("Wrote as %s" % output.name)
+
+def reduce_filename(f):
+    r'''
+    Expects something like /tmp/tmpAjry4Gdsbench/test.weights.e5.XXX.YYY.pb
+    Where XXX is a variation on the model size for example
+    And where YYY is a const related to the training dataset
+    '''
+
+    f = os.path.basename(f).split('.')
+    if f[1] == 'aot':
+        return 'AOT:' + str(keep_only_digits(f[-3]))
+    else:
+        return keep_only_digits(f[-3])
+
+def produce_plot_multiseries(input=None, output=None, title=None, size=None, fig_dpi=None, source_wav=None):
+    fig, ax = plt.subplots()
+    fig.set_figwidth(float(size.split('x')[0]) / fig_dpi)
+    fig.set_figheight(float(size.split('x')[1]) / fig_dpi)
+
+    nb_items = len(input[input.keys()[0]])
+    x_all    = range(nb_items)
+    for serie in input.keys():
+        serie_values = input[serie]
+
+        xtics  = map(lambda a: reduce_filename(a['model']), serie_values)
+        y      = map(lambda a: float(a['mean']), serie_values)
+        yerr   = map(lambda a: float(a['std']), serie_values)
+        linreg = scipy_stats.linregress(x_all, y)
+        ylin   = linreg.intercept + linreg.slope * numpy.asarray(x_all)
+
+        ax.errorbar(x_all, y, yerr=yerr, label=('%s' % serie), fmt='-', capsize=4, elinewidth=1)
+        ax.plot(x_all, ylin, label=('%s ~= %0.4f*x+%0.4f (R=%0.4f)' % (serie, linreg.slope, linreg.intercept, linreg.rvalue)))
+
+        plt.xticks(x_all, xtics, rotation=60)
+
+    if source_wav:
+        audio = wav.read(source_wav)
+        print('Adding realtime')
+        for rt_factor in [ 1.0, 1.5, 2.0 ]:
+            rt_secs = float(len(audio[1])) / float(1.0 * audio[0]) * rt_factor
```

Comment:
You don't need the `float()`s and the 1.0 stuff here if you import `__future__.division`.

Reply:
Done